### ISSUE TITLE
R Windows Error in duckdb::duckdb_register

### ISSUE THREAD

**Issue by phillc73** [cwida/duckdb#486]
Having installed duckdb 0.1.6 to specifically register virtual dataframes with `duckdb_register`, the function fails with:

`Error in duckdb_register(conn, view_name, value) : stoul`

Full code for reproduction copied from Issue #459 
```r
library("DBI")
con <- dbConnect(duckdb::duckdb())
df <- data.frame(col_1=c(1, 2, 3), 
	col_2=c("one", "two", "three"))

duckdb::duckdb_register(con, "df_view", df)

dbGetQuery(con, "SELECT * FROM df_view LIMIT 2")
```
R version 3.6.3
DBI version 1.1.0
Windows 10 Pro

This works fine on Linux.

**Comment by hannesmuehleisen** [comment]
Can you post the output of `sessionInfo()` please?

**Comment by phillc73** [comment]
```
> sessionInfo()
R version 3.6.3 (2020-02-29)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 18363)

Matrix products: default

locale:
[1] LC_COLLATE=English_Australia.1252  LC_CTYPE=English_Australia.1252
[3] LC_MONETARY=English_Australia.1252 LC_NUMERIC=C
[5] LC_TIME=English_Australia.1252

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base

loaded via a namespace (and not attached):
[1] compiler_3.6.3
```

**Comment by hannesmuehleisen** [comment]
And can you try to update your Rtools and try again?

**Comment by phillc73** [comment]
I do have the latest version of RTools installed, but your comment got me thinking.

I think the problem here is with RStudio.

I've tried running the provided example code in VSCode and directly in the R console. It successfully executes both times.

I'll dig a bit more into what could be happening with RStudio.

**Comment by hannesmuehleisen** [comment]
Interesting! I wonder where that difference comes from

**Comment by Pablo-Leon-INE** [comment]
Guess the same happening here:
### On Rstudio
> 
> library(DBI)
> 
> df <- data.frame(a=1:3, b=letters[1:3])
> 
> con <- dbConnect(duckdb::duckdb(), ":memory:")
> 
> dbWriteTable(con, "test", df, overwrite=TRUE, append=FALSE)
Error in duckdb_register(conn, view_name, value) : stoul
> 
> dbGetQuery(con, "select count(1) from test") 
  count(1)
1        0
> 
> dbDisconnect(con)
> 

### On R GUI
> library(DBI)
> 
> df <- data.frame(a=1:3, b=letters[1:3])
> 
> con <- dbConnect(duckdb::duckdb(), ":memory:")
> 
> dbWriteTable(con, "test", df, overwrite=TRUE, append=FALSE)
> 
> dbGetQuery(con, "select count(1) from test") 
  count(1)
1        3
> 
> dbDisconnect(con)
> 

### On Rstudio
> sessionInfo()
R version 3.6.3 (2020-02-29)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 18363)

Matrix products: default

locale:
[1] LC_COLLATE=Spanish_Chile.1252  LC_CTYPE=Spanish_Chile.1252    LC_MONETARY=Spanish_Chile.1252
[4] LC_NUMERIC=C                   LC_TIME=Spanish_Chile.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] DBI_1.1.0

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.3       dbplyr_1.4.2     packrat_0.5.0    crayon_1.3.4     dplyr_0.8.5     
 [6] assertthat_0.2.1 duckdb_0.1.6     R6_2.4.1         magrittr_1.5     pillar_1.4.3    
[11] rlang_0.4.5      rstudioapi_0.11  testthat_2.3.2   tools_3.6.3      glue_1.3.2      
[16] purrr_0.3.3      compiler_3.6.3   pkgconfig_2.0.3  tidyselect_1.0.0 tibble_2.1.3    

### On R GUI
> sessionInfo()
R version 3.6.3 (2020-02-29)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 18363)

Matrix products: default

Random number generation:
 RNG:     Mersenne-Twister 
 Normal:  Inversion 
 Sample:  Rounding 
 
locale:
[1] LC_COLLATE=Spanish_Chile.1252  LC_CTYPE=Spanish_Chile.1252   
[3] LC_MONETARY=Spanish_Chile.1252 LC_NUMERIC=C                  
[5] LC_TIME=Spanish_Chile.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] DBI_1.1.0

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.3       dbplyr_1.4.2     crayon_1.3.4     dplyr_0.8.5     
 [5] assertthat_0.2.1 duckdb_0.1.6     R6_2.4.1         magrittr_1.5    
 [9] pillar_1.4.3     rlang_0.4.5      testthat_2.3.2   glue_1.3.2      
[13] purrr_0.3.3      compiler_3.6.3   pkgconfig_2.0.3  tidyselect_1.0.0
[17] tibble_2.1.3

**Comment by hannesmuehleisen** [comment]
I think I know what causes this. Thanks for the detailed reports.

**Comment by hannesmuehleisen** [comment]
can you try again after reinstalling?

**Comment by Pablo-Leon-INE** [comment]
After reinstall:
## On Rstudio

> library(DBI)
> 
> df <- data.frame(a=1:3, b=letters[1:3])
> 
> con <- dbConnect(duckdb::duckdb(), ":memory:")
> 
> dbWriteTable(con, "test", df, overwrite=TRUE, append=FALSE)
> dbGetQuery(con, "select count(1) from test")
  count(1)
1        3
> dbDisconnect(con)

**We're good!**

> sessionInfo()
R version 3.6.3 (2020-02-29)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 18363)

Matrix products: default

locale:
[1] LC_COLLATE=Spanish_Chile.1252  LC_CTYPE=Spanish_Chile.1252    LC_MONETARY=Spanish_Chile.1252 LC_NUMERIC=C                  
[5] LC_TIME=Spanish_Chile.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] DBI_1.1.0

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.3       dbplyr_1.4.2     crayon_1.3.4     dplyr_0.8.5      assertthat_0.2.1 duckdb_0.1.6     R6_2.4.1         magrittr_1.5    
 [9] pillar_1.4.3     rlang_0.4.5      rstudioapi_0.11  testthat_2.3.2   tools_3.6.3      glue_1.3.2       purrr_0.3.3      compiler_3.6.3  
[17] pkgconfig_2.0.3  tidyselect_1.0.0 tibble_2.1.3  

* Is it OK that the change is not shown on the package version number?

**Comment by hannesmuehleisen** [comment]
Yes, I did not change the version number yet.

**Comment by hannesmuehleisen** [comment]
@phillc73 can you confirm and close if fixed? thanks!

**Comment by phillc73** [comment]
Apologies this took me a few days to get back to, but I confirm this is fixed for me in v0.1.7
